components/List.vue

```diff
@@ -6,97 +6,106 @@
+        <v-card height="100%">
+          <v-img height="40%" :src="article.urlToImage"></v-img>
+          <v-card-title
+            ><a :href="article.url">{{ article.title }}</a></v-card-title
+          >
+          <v-card-text>{{ article.author }}</v-card-text>
+          <v-card-text
+            ><h4>Published date :</h4>
+            20th January 2023</v-card-text
+          >
+          <v-btn @click="$router.push(`/details/${article.title}`)">
+            view Details
+          </v-btn>
+        </v-card>
+        <!-- </v-sheet> -->
       </v-col>
     </v-row>
   </v-container>
 </template>
-<script>
+
+<script lang="ts">
 import axios from 'axios'
-export default {
-  data() {
-    return {
-      artical: [],
-      page: 1,
-      pageSize: 10,
-      listCount: 0,
-      historyList: [],
-    }
-  },
-  async mounted() {
-    this.fetchData()
-  },
-  computed: {
-    pages() {
-      if (this.pageSize == null || this.listCount == null) return 0
-      return Math.ceil(this.listCount / this.pageSize)
+import Vue from 'vue'
+import { SingleDataType } from '~/interface/productInterface'
+export default Vue.extend(
+  {
+    data() {
+      let artical: Array<SingleDataType> = []
+      let page: number = 1
+      let pageSize: number = 10
+      let listCount: number = 0
+      let historyList: Array<SingleDataType> = []
+      let fetchData
+      return {
+        artical,
+        page,
+        pageSize,
+        listCount,
+        historyList,
+      }
     },
-  },
-  methods: {
-    async fetchData() {
-      await axios
-        .get(
-          'https://newsapi.org/v2/everything?q=bitcoin&apiKey=70905943afe0477ab21103fdbb396454'
-        )
-        .then((res) => {
-          this.artical = res.data.articles
-        })
-      this.initPage()
-      this.updatePage(this.page)
+    mounted() {
+      this.fetchData()
     },
-    initPage() {
-      this.listCount = this.artical.length
-      if (this.listCount < this.pageSize) {
-        this.historyList = this.artical
-      } else {
-        this.historyList = this.artical.slice(0, this.pageSize)
-      }
+    computed: {
+      pages(): number {
+        if (this.pageSize == null || this.listCount == null) return 0
+        return Math.ceil(this.listCount / this.pageSize)
+      },
     },
-    updatePage(pageIndex) {
-      let start = (pageIndex - 1) * this.pageSize
-      let end = pageIndex * this.pageSize
-      this.historyList = this.artical.slice(start, end)
-      this.page = pageIndex
-      console.log(pageIndex, this.pageSize, 'pageIndex')
-      console.log(this.historyList, 'historyList')
+    methods: {
+      async fetchData(): Promise<void> {
+        await axios
+          .get(
+            'https://newsapi.org/v2/everything?q=bitcoin&apiKey=70905943afe0477ab21103fdbb396454'
+          )
+          .then((res) => {
+            this.artical = res.data.articles
+          })
+        this.initPage()
+        this.updatePage(this.page)
+      },

```

Comment:
if you are getting data in .then API 
then what's the purpose of await

Reply:
Resolved